sql/sql_zip_dict.cc

```diff
@@ -94,8 +98,19 @@ int mysql_create_zip_dict(THD* thd, const char* name, ulong name_len,
         error= ER_READ_ONLY_MODE;
         my_error(error, MYF(0));
         break;
+      case HA_CREATE_ZIP_DICT_OUT_OF_MEMORY:
```

Comment:
Yes, just by reviewing the most typical resource allocation problems.
Test case  extended with checks for  ER_RECORD_FILE_FULL  as well as for ER_OUT_OF_RESOURCES.